[TEDSWS-314,359] Fix BT-67(b)-Procedure(-Language) ExclusionGround for v1.12+ w/ new vocab

schivmeister opened the pull request.

schivmeister committed:
Fix ExclusionGround type mapping for v1.12+, new vocab resource

The `dct:type` of an epo:ExclusionGround has used the `criterion`
controlled list (an EU Vocabulary "authority table", or AT) so far.
However, this is only applicable up to SDK v1.11.

From v1.12 onwards, a new AT `exclusion-ground` becomes applicable, that
is no longer a "sublist" of `criterion` and has the new `exg-` codes
used in newer notices.

Fixes TEDSWS-314.

schivmeister committed:
Fix BT-67(b)-Procedure(-Language) with versioned mapping for v1.12+

There was a change in the XPath for BT-67(b)-Procedure that goes along
with the vocabulary change for BT-67(a)-Procedure. This triggered a
general RMLMapper error on the language mapping function, as it matched
the older XPath, which expects a language tag.

In such a case the notice may have failed to get through the
transformation, as not all tools will be aware that output is still
produced despite the error. Updating the XPath for SDK v1.12+ yields a
non-match for current notices, and therefore the error will not trigger.

Note that in the event there are any valid XPaths that are expected to
come with a language attribute, such an error will still be generated.
The underlying issue will therefore need to be solved in an appropriate
way (adding guards, reporting upstream, error-handling in tools).

Note: The error manifests in notices like 336207-2025 for this specific
case, and some SDK v1.3 example notices for the general missing-language
case. Also, the language TriplesMap has been kept in its original
formatting, though not ideal, as we may need to be able to trace it back
in its original form at a later time.

Fixes TEDSWS-359.

csnyulas reviewed (approved):
Good. Nice fix for TEDSWS-314

schivmeister committed:
Fix empty ExclusionGround instances in PIN v1.12+

Move ExclusionGround subject TMap into versioned files, as the related
fields BT-67(a)-Procedure and BT-67(b)-Procedure have different XPaths.
There is now a listName attribute value `exclusion-ground-source` in
v1.12+ PIN notices, which is currently undefined in the SDK.

Catering for both cases in a base/shared TMap is problematic, as in new
notices the `@listName=exclusion-ground-source` is not matched, thereby
leading to runt ExclusionGround instances. Add appropriate conditions in
both versions to treat the situation consistently.

Additionally, remove an unwanted `rml:reference` from the BT-67(a) pOM,
and also adjust the label for these fields in v1.12+ according to the
SDK.
